### PR TITLE
add support to ConfigureMake for tweaking (first part of) test command via 'test_cmd'

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Review by Micket — APPROVED** [review]
lgtm

**Comment by ocaisa** [comment]
Test report by @ocaisa
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _LZO-2.10-GCCcore-11.2.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/368871719e7f00fa3787aaa8b0291342 for a full test report.

**Comment by ocaisa** [comment]
@boegel Interestingly, if I use this (with `test_cmd = 'echo'` in the ec) it works just fine:
```
== testing...
  >> running command:
        [started at: 2022-05-31 09:46:40]
        [working dir: /home/ocaisa/.local/easybuild/build/LZO/2.10/GCCcore-11.2.0/lzo-2.10]
        [output logged in /tmp/eb-0ij5uqg8/easybuild-run_cmd-0_0up1wg.log]
        echo test
  >> command completed: exit 0, ran in < 1s
```
but if I do the same thing in dry run mode, it reports the default:
```
[test_step method]
  >> running command:
	[started at: 2022-05-31 09:48:56]
	[working dir: /tmp/eb-baidba6y/__ROOT__/home/ocaisa/.local/easybuild/build/LZO/2.10/GCCcore-11.2.0/LZO-2.10]
	[output logged in /tmp/eb-baidba6y/easybuild-run_cmd-xlokny0u.log]
	make test
  running command "make test"
  (in /home/ocaisa/.local/easybuild/build/LZO/2.10/GCCcore-11.2.0/LZO-2.10)
```
That's because the easyblock git patch is not applied, which I think would perhaps not be expected. I'll open an issue for that.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _VEP-105-GCC-11.2.0.eb_ 
 * **SUCCESS** _BioPerl-1.7.8-GCCcore-10.3.0.eb_ 
 * **SUCCESS** _zlib-1.2.11-GCCcore-9.3.0.eb_ 
 * **SUCCESS** _wget-1.21.1-GCCcore-10.3.0.eb_ 
 * **SUCCESS** _protobuf-2.5.0-GCCcore-10.2.0.eb_ 
 * **SUCCESS** _OSU-Micro-Benchmarks-5.6.3-iimpi-2020b.eb_ 
 * **SUCCESS** _MPFR-4.1.0-GCCcore-10.3.0.eb_ 
 * **SUCCESS** _x265-3.5-GCCcore-11.2.0.eb_ 
 * **SUCCESS** _SeqAn-2.4.0-GCCcore-10.2.0.eb_ 
 * **SUCCESS** _protobuf-3.17.3-GCCcore-10.3.0.eb_ 
 * **SUCCESS** _openCARP-8.2-foss-2021a.eb_ 

Build succeeded for 11 out of 11 (11 easyconfigs in total)
node3141.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/bac36269febec41849dc1b07e73fdb93 for a full test report.

File: easybuild/easyblocks/generic/configuremake.py [hunk 1]
```diff
@@ -348,7 +350,12 @@ def test_step(self):
         """
 
         if self.cfg['runtest']:
```
Comment (Micket) [inline easybuild/easyblocks/generic/configuremake.py:352]: Hm, i just realized this leads the situation that you need to specify something in teh runtest part, else testing won't be considered. 
Perhaps change this to 
```python
if self.cfg['runtest'] or self.cfd['test_cmd'] != DEFAULT_TEST_CMD:
```

Comment (boegel) [reply easybuild/easyblocks/generic/configuremake.py:352]: @Micket Makes sense, please open a PR for that ASAP, maybe (just maybe) we can still include this into EasyBuild v4.5.5

Comment (boegel) [reply easybuild/easyblocks/generic/configuremake.py:352]: followed up in #2737

**Pull request merged**